woocommerce/woocommerce-android
# UI to select order statuses that will be used to filter order list.

Work in progress: #4934 

### Description
The task of filtering orders by order status was too big to create a single PR for it. In fact, this PR is already big enough. So I will split it in at least 2 PRs. This is the first part. 
In this PR I've added the essential UI to open filters flow in order list screen and select any combination of order statuses to filter. There are some things that are work in progress (I will indicate this in code) so no need to review them. 

Important pieces missing in this PR that will come in the following ones: 
- Loading order status from APIs call
- Persisting order status filters
- Update order list screen with selected filters
- Remove code related to order status from search functionality in order list screen

### Testing instructions
Open Order List and click the filters button. Play around selecting and unselecting different options, and check the filter screen is updated accordingly.  Behaviors to keep in mind. 

- Order status filters is multichoice
- When any order status is selected, "All" option should be unselected
- If "All" option si selected then all the rest of the selected filters should be unselected
- "Clear" button will wipe out all selections and will default to "All" option
- Back button will display a confirmation dialog before disscaring selected filters
